pkg/doc/verifiable/presentation.go

```diff
@@ -178,49 +179,58 @@ func (vp *Presentation) JWTClaims(audience []string, minimizeVP bool) *JWTPresCl
 	return newJWTPresClaims(vp, audience, minimizeVP)
 }
 
-// Credentials provides Verifiable Credentials enclosed into Presentation in raw byte array format.
-func (vp *Presentation) Credentials() ([]PresentationCredential, error) {
-	marshalSingleCredFn := func(cred interface{}) (PresentationCredential, error) {
-		credBytes, err := json.Marshal(cred)
-		if err != nil {
-			return nil, fmt.Errorf("marshal credentials from presentation: %w", err)
-		}
+// Credentials returns current credentials of presentation.
+func (vp *Presentation) Credentials() []interface{} {
+	return vp.credentials
+}
 
-		return credBytes, nil
+// SetCredentials defines credentials of presentation.
+// The credential could be string/byte (probably serialized JWT) or Credential structure.
+func (vp *Presentation) SetCredentials(creds ...interface{}) error {
```

Comment:
I'm a bit confused by this method... why not just do `vp.credentials = cred` directly wherever you call `SetCredentials`? For other types of credentials, do you think they won't assign creds to vp.creds? Perhaps you can clarify for me.

Reply:
The reason is to get control of what the users are passing.